[CI] DockerTests test150MachineDependentHeap failing

This failure started popping up on Jan 18 for the first time.

In this build scan, seems like the CLI tries to talk to ES, which exits unexpectedly while adding an index template. Exit code is 137, which our `TESTING.asciidoc` says means "out of memory".

Last logs from the test:
```
  1> {"@timestamp":"2024-01-26T05:57:56.178Z", "log.level": "INFO", "message":"adding index template [.monitoring-kibana-mb] for index patterns [.monitoring-kibana-8-*]", "ecs.version": "1.2.0","service.name":"ES_ECS","event.dataset":"elasticsearch.server","process.thread.name":"elasticsearch[c87ce626b1a7][masterService#updateTask][T#1]","log.logger":"org.elasticsearch.cluster.metadata.MetadataIndexTemplateService","elasticsearch.cluster.uuid":"Zav8rx__QLu6-mR0d8rhKw","elasticsearch.node.id":"zBWe0QaMQH2aG-coUKwcNw","elasticsearch.node.name":"c87ce626b1a7","elasticsearch.cluster.name":"docker-cluster"}

  1> Stderr:
  1> Jan 26, 2024 5:57:39 AM sun.util.locale.provider.LocaleProviderAdapter <clinit>
  1> WARNING: COMPAT locale provider will be removed in a future release

  1> ERROR: Elasticsearch exited unexpectedly, with exit code 137
  1> [2024-01-26T06:00:39,763][INFO ][o.e.p.u.Shell            ] [TEST-DockerTests.test150MachineDependentHeap-seed#[AA6947360780C1A7]-workerest_thread_info] Ran: [bash, -c, docker rm -f c87ce626b1a7a8472325f9aa69a2c24a170f7ddaa96ec12b6dc0779507b2c8e8] exitCode = [0] stdout = [c87ce626b1a7a8472325f9aa69a2c24a170f7ddaa96ec12b6dc0779507b2c8e8] stderr = []
```

No repro line + trying to run from within IntelliJ somehow skips the test, so couldn't repro locally.

**Build scan:**
https://gradle-enterprise.elastic.co/s/gznltc2u4khos/tests/:qa:packaging:destructiveDistroTest.default-docker/org.elasticsearch.packaging.test.DockerTests/test150MachineDependentHeap


**Reproduction line:**
```
null
```

**Applicable branches:**
8.12

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.packaging.test.DockerTests#test150MachineDependentHeap`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test150MachineDependentHeap'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.packaging.test.DockerTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: Failed to check whether Elasticsearch had started. This could be because authentication isn't working properly. Check the container logs

  at __randomizedtesting.SeedInfo.seed([AA6947360780C1A7:603C4726C5DB4757]:0)
  at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearch(Docker.java:589)
  at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearch(Docker.java:574)
  at org.elasticsearch.packaging.test.DockerTests.machineDependentHeapTest(DockerTests.java:1056)
  at org.elasticsearch.packaging.test.DockerTests.test150MachineDependentHeap(DockerTests.java:1015)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
  at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

  Caused by: java.lang.RuntimeException: Elasticsearch did not start

    at org.elasticsearch.packaging.util.ServerUtils.waitForElasticsearch(ServerUtils.java:292)
    at org.elasticsearch.packaging.util.docker.Docker.lambda$waitForElasticsearch$10(Docker.java:587)
    at org.elasticsearch.packaging.util.docker.Docker.withLogging(Docker.java:604)
    at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearch(Docker.java:587)
    at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearch(Docker.java:574)
    at org.elasticsearch.packaging.test.DockerTests.machineDependentHeapTest(DockerTests.java:1056)
    at org.elasticsearch.packaging.test.DockerTests.test150MachineDependentHeap(DockerTests.java:1015)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke(Method.java:580)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
    at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:1583)

    Caused by: javax.net.ssl.SSLHandshakeException: Remote host terminated the handshake

      at sun.security.ssl.SSLSocketImpl.handleEOF(SSLSocketImpl.java:1714)
      at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1514)
      at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1421)
      at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:455)
      at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:426)
      at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436)
      at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384)
      at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
      at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
      at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
      at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
      at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
      at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
      at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
      at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
      at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
      at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
      at org.apache.http.client.fluent.Request.internalExecute(Request.java:173)
      at org.apache.http.client.fluent.Executor.execute(Executor.java:262)
      at org.elasticsearch.packaging.util.ServerUtils.execute(ServerUtils.java:145)
      at org.elasticsearch.packaging.util.ServerUtils.waitForElasticsearch(ServerUtils.java:241)
      at org.elasticsearch.packaging.util.docker.Docker.lambda$waitForElasticsearch$10(Docker.java:587)
      at org.elasticsearch.packaging.util.docker.Docker.withLogging(Docker.java:604)
      at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearch(Docker.java:587)
      at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearch(Docker.java:574)
      at org.elasticsearch.packaging.test.DockerTests.machineDependentHeapTest(DockerTests.java:1056)
      at org.elasticsearch.packaging.test.DockerTests.test150MachineDependentHeap(DockerTests.java:1015)
      at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
      at java.lang.reflect.Method.invoke(Method.java:580)
      at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
      at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
      at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
      at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
      at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
      at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
      at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
      at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
      at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
      at java.lang.Thread.run(Thread.java:1583)

      Caused by: java.io.EOFException: SSL peer shut down incorrectly

        at sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:494)
        at sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:483)
        at sun.security.ssl.SSLSocketInputRecord.decode(SSLSocketInputRecord.java:160)
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:111)
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1506)
        at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1421)
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:455)
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:426)
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:436)
        at org.apache.http.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:384)
        at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:142)
        at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:376)
        at org.apache.http.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:393)
        at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:236)
        at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:186)
        at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:89)
        at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:110)
        at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:185)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:83)
        at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:56)
        at org.apache.http.client.fluent.Request.internalExecute(Request.java:173)
        at org.apache.http.client.fluent.Executor.execute(Executor.java:262)
        at org.elasticsearch.packaging.util.ServerUtils.execute(ServerUtils.java:145)
        at org.elasticsearch.packaging.util.ServerUtils.waitForElasticsearch(ServerUtils.java:241)
        at org.elasticsearch.packaging.util.docker.Docker.lambda$waitForElasticsearch$10(Docker.java:587)
        at org.elasticsearch.packaging.util.docker.Docker.withLogging(Docker.java:604)
        at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearch(Docker.java:587)
        at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearch(Docker.java:574)
        at org.elasticsearch.packaging.test.DockerTests.machineDependentHeapTest(DockerTests.java:1056)
        at org.elasticsearch.packaging.test.DockerTests.test150MachineDependentHeap(DockerTests.java:1015)
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.lang.reflect.Method.invoke(Method.java:580)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
        at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
        at java.lang.Thread.run(Thread.java:1583)

```

--- a/qa/packaging/src/test/java/org/elasticsearch/packaging/test/DockerTests.java
+++ b/qa/packaging/src/test/java/org/elasticsearch/packaging/test/DockerTests.java
@@ -1011,12 +1011,11 @@ public class DockerTests extends PackagingTestCase {
      * Check that when available system memory is constrained by Docker, the machine-dependant heap sizing
      * logic sets the correct heap size, based on the container limits.
      */
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/104786")
     public void test150MachineDependentHeap() throws Exception {
-        final List<String> xArgs = machineDependentHeapTest("942m", List.of());
+        final List<String> xArgs = machineDependentHeapTest("1536m", List.of());
 
-        // This is roughly 0.4 * 942
-        assertThat(xArgs, hasItems("-Xms376m", "-Xmx376m"));
+        // This is roughly 0.5 * 1536
+        assertThat(xArgs, hasItems("-Xms768m", "-Xmx768m"));
     }
 
     /**

